src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -505,6 +529,67 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     }
   }
 
+  override def deleteGoogleGroup(groupEmail: String): Unit = {
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+    val deleteGroupRequest = directoryService.groups.delete(groupEmail)
+    Try(executeGoogleRequest(deleteGroupRequest)) match {
+      case Failure(_) => logger.warn(s"Failed to delete group $groupEmail")
+      case Success(_) =>
+    }
+  }
+
+  /**
+    * create a new Google group
+    * @param groupEmail     the name of the new Google group to be created. it must be formatted as an email address
+    * @return               Option of the groupEmail address if all above is successful, otherwise if the group creation
+    *                        step failed, return Option.empty
+    */
+  override def createGoogleGroup(groupEmail: String): Option[String] = {
+    val newGroup = new Group().setEmail(groupEmail)
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+
+    val insertRequest = directoryService.groups.insert(newGroup)
+    Try(executeGoogleRequest[Group](insertRequest)) match {
+      case Failure(response) => {
+        val errorCode = response.asInstanceOf[GoogleJsonResponseException].getDetails.getCode
```

Comment:
Can this throw exceptions other than `GoogleJsonResponseException`? A safer way to handle exceptions may be something like:

```
case Failure(response: GoogleJsonResponseException) => {
   //your existing error handling here
}
case Failure(_) => {
  //handle a generic exception here
}
case Success(newGroupInfo) => {
  //success case here
}
```

I'm not in front of a Scala IDE right now (reviewing from personal laptop) so I can't actually try this out but it _should_ allow you to get rid of the `asInstanceOf`s which could cause problems if another exception is ever thrown.

If this works, I'd also apply it to the other areas where you're using `.asInstanceOf` in error handling (except for tests, using it there is fine).

Reply:
awesome, great catch. I made this change (awesome that it removes the need to use `asInstanceOf`!) in both `createGoogleGroup` and `addMemberToAnonymizedGoogleGroup`.